src/ucp/wireup/wireup.c

```diff
@@ -1029,6 +1103,103 @@ ucp_wireup_get_reachable_mds(ucp_ep_h ep,
     key->reachable_md_map = dst_md_map;
 }
 
+static void
+ucp_wireup_ep_config_intersect(const ucp_ep_config_key_t *old_key,
+                               const ucp_ep_config_key_t *new_key,
+                               ucp_lane_index_t *intersect_map)
+{
+    ucp_lane_index_t old_lane_idx;
+
+    for (old_lane_idx = 0; old_lane_idx < old_key->num_lanes; ++old_lane_idx) {
+        /* Go through the old configuration and check if the lanes selected for
+         * the old configuration could be used for the new configuration */
+        intersect_map[old_lane_idx] =
+            ucp_wireup_ep_configs_can_reuse_lane(old_key, new_key, old_lane_idx);
+    }
+}
+
+static ucp_lane_index_t
+ucp_wireup_ep_config_find_intersect_lane(const ucp_ep_config_key_t *old_key,
+                                         const ucp_lane_index_t *intersect_map,
+                                         ucp_lane_index_t new_lane)
```

Comment:
pls abstract away from usage, no need new/old prefixes, you just do value lookup in array